src/main/java/bio/terra/workspace/service/workspace/flight/StoreGoogleBucketMetadataStep.java

```diff
@@ -0,0 +1,45 @@
+package bio.terra.workspace.service.workspace.flight;
+
+import bio.terra.stairway.FlightContext;
+import bio.terra.stairway.FlightMap;
+import bio.terra.stairway.Step;
+import bio.terra.stairway.StepResult;
+import bio.terra.stairway.exception.RetryException;
+import bio.terra.workspace.db.ControlledResourceDao;
+import bio.terra.workspace.service.controlledresource.model.ControlledResourceMetadata;
+import java.util.UUID;
+
+public class StoreGoogleBucketMetadataStep implements Step {
+
+  private final ControlledResourceDao controlledResourceDao;
+
+  public StoreGoogleBucketMetadataStep(ControlledResourceDao controlledResourceDao) {
+    this.controlledResourceDao = controlledResourceDao;
+  }
+
+  @Override
+  public StepResult doStep(FlightContext flightContext)
+      throws InterruptedException, RetryException {
+    final FlightMap workingMap = flightContext.getWorkingMap();
+    final ControlledResourceMetadata controlledResourceMetadata =
+        ControlledResourceMetadata.builder()
```

Comment:
I think we should build the intermediate structure in the controller - not passing the request body through. That is the general practice.

If you did that AND you didn't use @AutoValue to make it, you could pass that structure in on the JobBuilder as the request parameter. That parameter becomes the REQUEST entry in the inputParameters FlightMap. That would save all of this to-ing and fro-ing with individual parameters.

I think some of these parameters are from the inputParameter map and not from the working map.